plugin/trino-openlineage/src/main/java/io/trino/plugin/openlineage/OpenLineageListener.java

```diff
@@ -0,0 +1,370 @@
+                                        .name(queryCreatedEvent.getMetadata().getQueryId())
+                                        .facets(
+                                                ol.newJobFacetsBuilder()
+                                                        .sql(ol.newSQLJobFacet(queryCreatedEvent.getMetadata().getQuery()))
+                                                        .build())
+                                        .build())
+                        .build();
+
+        return startEvent;
+    }
+
+    public OpenLineage.RunEvent getCompletedEvent(UUID runID, QueryCompletedEvent queryCompletedEvent)
+            throws IllegalAccessException
+    {
+        boolean failed = queryCompletedEvent.getMetadata().getQueryState().equals("FAILED");
+
+        OpenLineage.RunFacetsBuilder runFacetsBuilder = ol.newRunFacetsBuilder();
+        runFacetsBuilder.put(OpenLineageTrinoFacet.TRINO_METADATA.getText(),
+                getTrinoMetadataFacet(queryCompletedEvent.getMetadata()));
+        runFacetsBuilder.put(OpenLineageTrinoFacet.TRINO_QUERY_CONTEXT.getText(),
+                getTrinoQueryContextFacet(queryCompletedEvent.getContext()));
+        runFacetsBuilder.put(OpenLineageTrinoFacet.TRINO_QUERY_STATISTICS.getText(),
+                getTrinoQueryStatisticsFacet(queryCompletedEvent.getStatistics()));
+
+        OpenLineage.RunEvent completedEvent =
+                ol.newRunEventBuilder()
+                        .eventType(
+                                failed
+                                        ? OpenLineage.RunEvent.EventType.FAIL
+                                        : OpenLineage.RunEvent.EventType.COMPLETE)
+                        .eventTime(queryCompletedEvent.getEndTime().atZone(ZoneId.of("UTC")))
+                        .run(ol.newRunBuilder().runId(runID).facets(runFacetsBuilder.build()).build())
+                        .job(
+                                ol.newJobBuilder()
+                                        .namespace(this.jobNamespace)
+                                        .name(queryCompletedEvent.getMetadata().getQueryId())
+                                        .facets(
+                                                ol.newJobFacetsBuilder()
+                                                        .sql(ol.newSQLJobFacet(queryCompletedEvent.getMetadata().getQuery()))
+                                                        .build())
+                                        .build())
+                        .inputs(buildInputs(queryCompletedEvent.getIoMetadata()))
+                        .outputs(buildOutputs(queryCompletedEvent.getIoMetadata()))
+                        .build();
+
+        return completedEvent;
+    }
+
+    private List<OpenLineage.InputDataset> buildInputs(QueryIOMetadata ioMetadata)
+    {
+        return ioMetadata.getInputs().stream().map(inputMetadata ->
+                ol.newInputDatasetBuilder()
```

Comment:
As per dataset naming conventions
https://openlineage.io/docs/spec/naming/#dataset-naming

Shouldn't 
`namespace` be set as `{host}:{port}`
and `name` as `{catalog}.{schema}.{table}` here ?

Reply:
<img width="1055" alt="Screenshot 2024-04-15 at 7 44 38 AM" src="https://github.com/trinodb/trino/assets/110063541/841172e1-8413-4a02-9425-9d5624df173a">

Reply:
sure, will align it. although this PR preceds refactor of dataset naming, so I would rather finally merge this rather than wait for continuous changes in OL conventions :)